src/openrct2/rct2/S6Importer.cpp

```diff
@@ -169,6 +169,11 @@ class S6Importer final : public IParkImporter
         return false;
     }
 
+    bool HasInvalidObjects() override
```

Comment:
You called this function  `HasInvalidObjects()`, yet it returns whether the save contains _no_ invalid objects. The functions it calls all return true on success anyway (i.e. all valid), so I'd rename this function to `HasNoInvalidObjects()` or something similar.